spec/builtin/net/cert_enum_spec.rb

```diff
@@ -3,38 +3,63 @@
 
 describe Ronin::Recon::Net::CertEnum do
   describe "#process" do
-    context "when there are Values in cert" do
-      let(:fixtures_dir)  { File.expand_path(File.join(__dir__,'..','..','values','fixtures')) }
-      let(:cert_path)     { File.join(fixtures_dir,'example.crt') }
-      let(:cert)          { Ronin::Support::Crypto::Cert.load_file(cert_path) }
-      let(:expected) do
-        %w[
-          www.example.org
-          www.example.org
-          www.example.com
-          www.example.edu
-          www.example.net
-          example.org
-          example.com
-          example.edu
-          example.net
-        ]
+    context "when there are values in cert" do
+      context "with subject alt names" do
+        let(:fixtures_dir)  { File.expand_path(File.join(__dir__,'..','..','values','fixtures')) }
+        let(:cert_path)     { File.join(fixtures_dir,'example.crt') }
+        let(:cert)          { Ronin::Support::Crypto::Cert.load_file(cert_path) }
+        let(:expected) do
+          %w[
+            www.example.org
+            www.example.org
+            www.example.com
+            www.example.edu
+            www.example.net
+            example.org
+            example.com
+            example.edu
+            example.net
+          ]
+        end
+
+        it "must yield Values found in subject and subject_alt_names" do
+          yielded_values = []
+
+          subject.process(cert) do |value|
+            yielded_values << value
+          end
+
+          expect(yielded_values).to_not be_empty
+          expect(yielded_values.map(&:name)).to match_array(expected)
+        end
       end
 
-      it "must yield found Values" do
-        yielded_values = []
+      context "without subject alt names" do
+        let(:key)          { Ronin::Support::Crypto::Key::RSA.random }
+        let(:cert)         { Ronin::Support::Crypto::Cert.generate(key:, subject: x509_subject) }
```

Comment:
You could create a new `Ronin::Support::Crypto::Cert.new` object and then set `cert.subject = x509_subject`, after populating `x509_subject`. This won't require generating a signing key or signing the cert.